ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -370,6 +356,23 @@ empty fetch range, but this is ok since we never request empty ranges.
 newtype ChainSuffix header =
     ChainSuffix { getChainSuffix :: AnchoredFragment header }
 
+data BadChainSuffix = ChainSuffixEmpty | ChainSuffixNoIntersection
+  deriving (Eq, Ord, Show)
+
+-- | Smart constructor, enforcing 'ChainSuffix' invariants
+mkChainSuffix
+  :: (HasHeader header, HasHeader block,
+      HeaderHash header ~ HeaderHash block)
+  => AnchoredFragment block  -- ^ Current chain.
+  -> AnchoredFragment header -- ^ Candidate chain
+  -> Either BadChainSuffix (ChainSuffix header)
```

Comment:
Consider a new name for `ChainSuffix`, something more direct, such as `NewBlocks`.